sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/EndToEndTimeOutValidationTests.java

```diff
@@ -0,0 +1,373 @@
+
+        StepVerifier.create(queryPagedFlux)
+            .expectErrorMatches(throwable -> throwable instanceof RequestCancelledException)
+            .verify();
+    }
+
+    @Test(groups = {"simple"}, timeOut = 10000L)
+    public void clientLevelEndToEndTimeoutPolicyInOptionsShouldTimeout() {
+        if (getClientBuilder().buildConnectionPolicy().getConnectionMode() != ConnectionMode.DIRECT) {
+            throw new SkipException("Failure injection only supported for DIRECT mode");
+        }
+
+        CosmosClientBuilder builder = new CosmosClientBuilder()
+            .endpoint(TestConfigurations.HOST)
+            .endToEndOperationLatencyPolicyConfig(endToEndOperationLatencyPolicyConfig)
+            .credential(credential);
+
+        try (CosmosAsyncClient cosmosAsyncClient = builder.buildAsyncClient()) {
+            String dbname = "db_" + UUID.randomUUID();
+            String containerName = "container_" + UUID.randomUUID();
+            CosmosContainerProperties properties = new CosmosContainerProperties(containerName, "/mypk");
+            cosmosAsyncClient.createDatabaseIfNotExists(dbname).block();
+            cosmosAsyncClient.getDatabase(dbname)
+                .createContainerIfNotExists(properties).block();
+            CosmosAsyncContainer container = cosmosAsyncClient.getDatabase(dbname)
+                .getContainer(containerName);
+
+            TestObject obj = new TestObject(UUID.randomUUID().toString(),
+                "name123",
+                2,
+                UUID.randomUUID().toString());
+            CosmosItemResponse response = container.createItem(obj).block();
+
+            Mono<CosmosItemResponse<TestObject>> cosmosItemResponseMono =
+                container.readItem(obj.id, new PartitionKey(obj.mypk), TestObject.class);
+
+            // Should read item properly before injecting failure
+            StepVerifier.create(cosmosItemResponseMono)
+                .expectNextCount(1)
+                .expectComplete()
+                .verify();
+
+            injectFailure(container, FaultInjectionOperationType.READ_ITEM, null);
+
+            // Should timeout after injected delay
+            verifyExpectError(cosmosItemResponseMono);
+
+            String queryText = "select top 1 * from c";
+            SqlQuerySpec sqlQuerySpec = new SqlQuerySpec(queryText);
+
+            CosmosPagedFlux<TestObject> queryPagedFlux = container.queryItems(sqlQuerySpec, TestObject.class);
+
+            // Should query item properly before injecting failure
+            StepVerifier.create(queryPagedFlux)
+                .expectNextCount(1)
+                .expectComplete()
+                .verify();
+
+            injectFailure(container, FaultInjectionOperationType.QUERY_ITEM, null);
+
+            // Should timeout after injected delay
+            StepVerifier.create(queryPagedFlux)
+                .expectErrorMatches(throwable -> throwable instanceof RequestCancelledException)
```

Comment:
should we also validate the cosmosDiagnostics here?

Reply:
As discussed, Will deal with diagnostic and validating diagnostics separately.